src/django/api/views.py

```diff
@@ -554,6 +569,131 @@ def retrieve(self, request, pk=None):
         except Facility.DoesNotExist:
             raise NotFound()
 
+    @transaction.atomic
+    def destroy(self, request, pk=None):
+        if request.user.is_anonymous:
+            raise NotAuthenticated()
+        if not request.user.is_superuser:
+            raise PermissionDenied()
+
+        try:
+            facility = Facility.objects.get(pk=pk)
+        except Facility.DoesNotExist:
+            raise NotFound()
+
+        if facility.get_approved_claim():
+            raise BadRequestException(
+                'Facilities with approved claims cannot be deleted'
+            )
+
+        now = str(datetime.utcnow())
+        list_item = facility.created_from
+        list_item.status = FacilityListItem.DELETED
+        list_item.processing_results.append({
+            'action': ProcessingAction.DELETE_FACILITY,
+            'started_at': now,
+            'error': False,
+            'finished_at': now,
+            'deleted_oar_id': facility.id,
+        })
+        list_item.facility = None
+        list_item.save()
+
+        match = facility.get_created_from_match()
+        match.changeReason = 'Deleted {}'.format(facility.id)
```

Comment:
Out of curiosity, is there a reason that `changeReason` is in camel case rather than snake case?

Reply:
(I may have missed this in a prior migration. I think it's probably fine to leave; just wanted to be deliberate about having it.)

Reply:
Oh -- I just realized that we got this from django-simple-history. Interesting choice on the library's part but I guess it could be a way to signal that it's a slightly different kind of field.

Reply:
Yeah. It hurts a little :-)

Strange choice by the library author(s).